request_test.go

```diff
@@ -1237,6 +1238,76 @@ func TestRequest_BodyMultipart(t *testing.T) {
 		eof, _ := reader.NextPart()
 		assert.Nil(t, eof)
 	})
+
+	t.Run("multipart writer error", func(t *testing.T) {
+		cases := []struct {
+			name     string
+			reqFn    func() *Request
+			assertFn func(t *testing.T, req *Request)
+		}{
+			{
+				name: "with form",
+				reqFn: func() *Request {
+					req := NewRequestC(config, "POST", "url")
+					req.multipartFn = func(w io.Writer) *multipart.Writer {
+						return multipart.NewWriter(&failureWriter{})
+					}
+					req.WithMultipart()
+					req.WithForm(map[string]string{"foo": "bar"})
+					return req
+				},
+				assertFn: func(t *testing.T, req *Request) {
+					req.chain.assertFailed(t)
+				},
```

Comment:
Seems this part:

```go
				assertFn: func(t *testing.T, req *Request) {
					req.chain.assertFailed(t)
				},
```

is same for all subtests, so we can remove assertFn and move the code to the loop.

Also please use assertOK(t, false) instead of assertFailed(t), I've started a (slow) process of migrating from assertFailed and assertNotFailed to assertOK.

Reply:
updated https://github.com/gavv/httpexpect/pull/344/commits/04042a6e44d71805c60bd675cf13ff067f5db142